src/nested_pandas/nestedframe/core.py

```diff
@@ -154,3 +157,134 @@ def query(self, expr) -> Self:  # type: ignore[name-defined] # noqa: F821
+            `nested_df` refers to a particular nested dataframe and
+            `nested_col` is a column of that nested dataframe).
+        inplace : bool, default False
+            Whether to modify the DataFrame rather than creating a new one.
+        ignore_index : bool, default ``False``
+            If ``True``, the resulting axis will be labeled 0, 1, …, n - 1.
+
+            .. versionadded:: 2.0.0
+
+        Returns
+        -------
+        DataFrame or None
+            DataFrame with NA entries dropped from it or None if ``inplace=True``.
+
+        Notes
+        -----
+        Operations that target a particular nested structure return a dataframe
+        with rows of that particular nested structure affected.
+        """
+
+        # determine target dataframe
+
+        # first check the subset kwarg input
+        subset_target = []
+        if subset:
+            if isinstance(subset, str):
+                subset = [subset]
+
+            for col in subset:
+                col = col.split(".")[0] if "." in col else col
+                if col in self.nested_columns:
+                    subset_target.append(col)
+                elif col in self.columns:
+                    subset_target.append("base")
+
+            # import pdb;pdb.set_trace()
+
+            # Check for 1 target
+            subset_target = np.unique(subset_target)
+            if len(subset_target) > 1:  # prohibit multi-target operations
+                raise ValueError(
+                    f"Targeted multiple nested structures ({subset_target}), write one command per target dataframe"  # noqa
+                )
+            elif len(subset_target) == 0:
+                raise ValueError(
+                    "Provided base columns or nested layer did not match any found in the nestedframe"
+                )
+            subset_target = subset_target[0]
+
+        # Next check the on_nested kwarg input
+        if on_nested and on_nested not in self.nested_columns:
+            raise ValueError("Provided nested layer not found in nested dataframes")
+
+        # Resolve target layer
+        target = "base"
+        if on_nested and subset_target:
+            if on_nested != subset_target:
+                raise ValueError(
+                    f"Provided on_nested={on_nested}, but subset columns are from {subset_target}. Make sure these are aligned or just use subset."  # noqa
+                )
+            else:
+                target = subset_target
+        elif on_nested:
+            target = str(on_nested)
+        elif subset_target:
+            target = str(subset_target)
+
+        if target == "base":
+            return super().dropna(
+                axis=axis, how=how, thresh=thresh, subset=subset, inplace=inplace, ignore_index=ignore_index
+            )
+        else:
+            if subset is not None:
+                subset = [col.split(".")[-1] for col in subset]
+            self[target] = packer.pack_flat(
```

Comment:
We kinda ignore `inplace` for the `self` here. Should we also create a new `NestedFrame` for `inplace=False`?

Reply:
See the latest commit for a failing test

Reply:
Failed test was due to some bad sequencing of dropna calls, found that inplace=True was failing further in the tests, and have fixed it as well.